plenum/server/catchup/node_leecher_service.py

```diff
@@ -77,6 +81,7 @@ def start(self, is_initial: bool = False):
             leecher.reset()
 
         self._catchup_till.clear()
+        self._nodes_ledger_sizes.clear()
```

Comment:
A side note:
The more mutable states we add to these service, the more I think that it would be great to instantiate it for every catchup and make it more immutable.

Reply:
Agreed, however this could be quite risky due to services inter-dependencies and how Python manages memory.